### DISCUSSION TITLE
How do I change things

### DISCUSSION THREAD

**Discussion by Toolfolks** [souzatharsis/podcastfy#90] [General]
1. Change open AI voices. 
2. Can I use more than 2 voices.
3. "conversation_style": ["casual", "humorous"], what are the parameters available.
4. What is the correct syntax for local PDF ?

Cheers

**Comment by souzatharsis** [comment]
Hi,

2. No. Currently only a two person conversation is possible.

See how to guide for 1 and 3.
https://github.com/souzatharsis/podcastfy/blob/main/usage/how-to.md

4. There is a bug on pdf to podcast generation. I'll fix it when I'm back
to keyboard in a couple of days. Apologies for that.

**Comment by Toolfolks** [comment]
I'm struggling with the syntax

`  openai:
    default_voices:
      question: "onyx"
      answer: "echo"
    model: "tts-1-hd"`


In app.py

`custom_config = {
    "word_count": 300,
    "conversation_style": ["casual", "humorous"],
    "podcast_name": "Steve Warbys Info Podcast",
    "creativity": 0.7,
    "default_question_voice": "onyx",  # Previously under openai.default_voices.question
    "default_answer_voice": "echo",     # Previously under openai.default_voices.answer
    "tts_model": "tts-1-hd"   
    }`
    
    What is the correct syntax please.
    
    Also when using my app.py how do I set it to use 
    
    custom_config.yaml
    
    
Cheers

**Comment by souzatharsis** [comment]
In podcastfy.ipynb there is a customization section where it shows how to
pass conversation config as a dictionary in python application.

**Reply by Toolfolks** [reply]
I have the following custom_config.yaml

`#word_count: 2000
conversation_style: 
  - "engaging"
  - "fast-paced"
  - "enthusiastic"
roles_person1: "main summarizer"
roles_person2: "questioner/clarifier"
dialogue_structure: 
  - "Introduction"
  - "Main Content Summary"
  - "Conclusion"
#podcast_name: "PODCASTFY"
podcast_tagline: "Listen While you work"
output_language: "English"
engagement_techniques: 
  - "rhetorical questions"
  - "anecdotes"
  - "analogies"
  - "humor"
creativity: 1

text_to_speech:
  default_tts_model: "openai"
  elevenlabs:
    default_voices:
      question: "Chris"
      answer: "Jessica"
    model: "eleven_multilingual_v2"
  openai:
    default_voices:
      question: "onyx"
      answer: "onyx"
    model: "tts-1-hd"
  edge:
    default_voices:
      question: "en-US-JennyNeural"
      answer: "en-US-EricNeural"
  audio_format: "mp3"
  temp_audio_dir: "data/audio/tmp/"
  ending_message: "Bye Bye!"`
  
  app.py
  
  `from podcastfy.client import generate_podcast

custom_config = {
    "word_count": 100,
    "conversation_style": ["casual", "humorous"],
    "podcast_name": "Steve Warbys Info Podcast",
    "creativity": 0.7
    
    }



generate_podcast(

    urls =["https://www.dewalt.co.uk/products/storage","https://www.dewalt.co.uk/products/storage/backpacks","https://www.dewalt.co.uk/products/storage/mobile-tool-storage","https://www.dewalt.co.uk/products/storage/storage-combo-kits","https://www.dewalt.co.uk/products/storage/tool-bags-pouches-belts","https://www.dewalt.co.uk/products/storage/tool-organizers-bins","https://www.dewalt.co.uk/products/storage/trolleys","https://www.dewalt.co.uk/products/storage/workshop-storage"],
    
    conversation_config=custom_config
)`

The generated voices are not changed.

What am I not understanding here please

**Comment by Toolfolks** [comment]
I have worked out that it uses conversation_config.yaml to set the defaults.
Also I can change some of these with 

custom_config = {
    "word_count": 6000,
    "conversation_style": ["casual", "humorous"],
    "podcast_name": "Steve Warbys Info Podcast"
    
    }
    
    conversation_config=custom_config,
    
    How do I get it to use story_custom.yaml settings ?

**Comment by pedrobfig** [comment]
Hey everyone.

https://cloud.google.com/text-to-speech

Any chance or suggestion to use Google TTS? As far as I can see, pricing is much more interesting. They even offer the first million characters every month. We can also adjust pitch and speed of voices, something that I would like to adjust for different types of podcasts.

We can find there those two amazing voices from NotebookLM. ;-)

Cheers.

**Comment by souzatharsis** [comment]
Yes, there's a PR from a contributor that implemented exsctky this. I'll
merge ASAP.

Cutious: What are the two voices from notebooklm you found? Are those the
journey voices?

**Reply by pedrobfig** [reply]
Hi again.

You can find them as "Journey" voices:

https://cloud.google.com/text-to-speech/docs/voice-types

Journey Voices, powered by the AudioLM engine, lets you create more engaging and empathetic speech for conversational applications.

You should also check "Casual" voices, that are very similar to the NotebookLM voices.

**Comment by crossterra-agent** [comment]
- **Status**: DONE
- **Role**: Orchestrator
- **Link**: https://github.com/CrossTerra/monorepo/pull/1953
- **Summary**: Created PR with 2-step implementation plan; developer invoked to implement cancel order button on order details page.
